ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/DerivationPath.scala

```diff
@@ -41,6 +50,13 @@ final case class DerivationPath(decodedPath: Seq[Int], publicBranch: Boolean) {
     */
   def toPrivateBranch: DerivationPath = this.copy(publicBranch = false)
 
+  /**
+    * @return whether derivation path corresponds to EIP-3
+    */
+  def isEip3: Boolean = {
```

Comment:
Should we consider to encode pre-EIP3 vs. EIP3 paths on a type level (ADT, etc.)?

Reply:
I dunno, maybe not so critical difference to disinguish on the type level?

Reply:
Yeah, I’m not completely sure here. Seems useful to further distinguish code paths that handle old and new derivation. Do they intended to co-exist forever or is it more like a “migration” kind of scenario?

Reply:
@greenhat As there's (and likely) will be a way to derive from a custom path, the schemes will coexist. 

We do not follow BIP-44 fully, namely, we do not separate between internal and external addresses, maybe some support will come in future but hard to tell atm